src/syscheckd/whodata/syscheck_audit.c

```diff
@@ -179,23 +171,93 @@ int init_auditd_socket(void) {
     return sfd;
 }
 
-void audit_no_rules_to_realtime() {
-    int found;
+void audit_create_rules_file() {
     char *real_path = NULL;
     int i;
+    FILE *fp;
+
+    fp = fopen(AUDIT_RULES_FILE, "w");
+    if (!fp) {
+        merror(FOPEN_ERROR, AUDIT_RULES_FILE, errno, strerror(errno));
+        return;
+    }
 
     for (i = 0; syscheck.dir[i] != NULL; i++) {
         if ((syscheck.opts[i] & WHODATA_ACTIVE) == 0) {
             continue;
         }
         real_path = fim_get_real_path(i);
-        found = search_audit_rule(real_path, WHODATA_PERMS, AUDIT_KEY);
 
-        if (found == 0) {   // No rule found
+        mdebug2(FIM_ADDED_RULE_TO_FILE, real_path);
+        fprintf(fp, "-w %s -p wa -k %s\n", real_path, AUDIT_KEY);
+
+        free(real_path);
+    }
+
+    if (fclose(fp)) {
+        merror(FCLOSE_ERROR, AUDIT_RULES_FILE, errno, strerror(errno));
+        return;
+    }
+
+    // Create symlink to audit rules file
+    if (symlink(AUDIT_RULES_FILE, AUDIT_RULES_LINK) < 0) {
+        switch (errno) {
+        case EEXIST:
+            if (unlink(AUDIT_RULES_LINK) < 0) {
+                merror(UNLINK_ERROR, AUDIT_RULES_LINK, errno, strerror(errno));
+                return;
+            }
+
+            if (symlink(AUDIT_RULES_FILE, AUDIT_RULES_LINK) == 0) {
+                break;
+            }
+
+        // Fallthrough
+        default:
+            merror(LINK_ERROR, AUDIT_RULES_LINK, AUDIT_RULES_FILE, errno, strerror(errno));
+            return;
+        }
```

Comment:
I think using a ifs here could improve readability
```suggestion
        if (errno != EEXIST) {
            merror(LINK_ERROR, AUDIT_RULES_LINK, AUDIT_RULES_FILE, errno, strerror(errno));
            return;
        }

        if (unlink(AUDIT_RULES_LINK) < 0) {
            merror(UNLINK_ERROR, AUDIT_RULES_LINK, errno, strerror(errno));
            return;
        }

        if (symlink(AUDIT_RULES_FILE, AUDIT_RULES_LINK) < 0) {
            merror(LINK_ERROR, AUDIT_RULES_LINK, AUDIT_RULES_FILE, errno, strerror(errno));
            return;
        }
```